docs_app/content/tutorial/3-creating-an-observable.md

```diff
@@ -0,0 +1,140 @@
+        // Push N values to your consumer with `subscriber.next`
+        subscriber.next(data);
+    };
+
+    dataService.onerror = (err) => {
+        // if you have an error, notify the consumer with `subscriber.error`
+        subscriber.error(err);
+    };
+
+    dataService.onclose = () => {
+        // If the producer is done pushing values, notify the consumer
+        // with `subscriber.complete`
+        subscriber.complete();
+    };
+
+    return () => {
+        // Teardown logic goes here
+        dataService.destroy();
+    };
+});
+```
+
+### Synchronous "firehoses"
+
+Generally, it is recommended that you create observables that are asynchronous. However, sometimes it is necessary to create an observable that may need to deliver messages synchronously. Generally, there is no difference in how you would use the `Observable` [constructor](API), with the exception of what we call a "synchronous firehose".
+
+A "synchronous firehose" is a data producer that will synchronously provide a very large number of values very quickly. In this case, you'll need to make sure you're able to stop whatever is consuming this "firehose" as soon as it is no longer necessary to push values.
+
+This is tricky to explain. If an observable can deliver messages synchronously when you call [`subscribe`](API), and it delivers this "firehose", how could a [consumer](GL) [unsubscribe](GL) from it? You don't get the [`Subscription`](API) object until [`subscribe`](API) returns! Well, the answer is operators like [`take`](API), [`takeWhile`](API), [`takeUntil`](API), and [`first`](API). We haven't started discussing operators quite yet, but they're important to consider when constructing a "synchronous firehose" type observable, because they can cause unsubscription during notification.
+
+Let's take a look at the most basic "synchronous firehose":
+
+```ts
+// BAD: this is going to lock your thread on subscribe!
+const firehose$ = new Observable(subscriber => {
+    let n = 0;
+    while (true) {
+        subscriber.notify(n++);
+    }
+});
+
+firehose$.pipe(
+    // We only wanted (and will only get) 10 values!
+    take(10)
+)
+.subscribe(console.log); // Doom here.
+```
+
+The code above will indeed only log 10 values, but it will still lock your thread. So what gives? The answer may seem obvious: There's nothing to stop that `while` loop in the event that the `subscriber` can no longer push values to the [consumer](GL).
+
+#### Subscriber closed
+
+Synchronously, when a subscriber has called `complete` or `error`, or when a consumer has [unsubscribed](GL), there is a flag property on [`Subscriber`](API) called [`closed`](API) that will be flipped to `true`. This is useful for checking whether to continue things like synchronous loops. (NOTE: That you do _NOT_ have to check `closed` every time before calling `next`, `error`, or `complete`, RxJS does an internal check there for you).
+
+```ts
+// BAD: this is going to lock your thread on subscribe!
```

Comment:
I'm confused. This seems like the way to fix the above bad example, but it's till commented as BAD. Also, the previous paragraph says that you do not have to check closed before calling next, but this example precisely does that.